substrate/frame/bounties/src/lib.rs

```diff
@@ -808,6 +824,52 @@ pub mod pallet {
 			Self::deposit_event(Event::<T, I>::BountyExtended { index: bounty_id });
 			Ok(())
 		}
+
+		/// Approve bountry and propose a curator simultaneously.
+		/// This call is a shortcut to calling `approve_bounty` and `propose_curator` separately.
+		///
+		/// May only be called from `T::SpendOrigin`.
+		///
+		/// - `bounty_id`: Bounty ID to approve.
+		/// - `curator`: The curator account whom will manage this bounty.
+		/// - `fee`: The curator fee.
+		///
+		/// ## Complexity
+		/// - O(1).
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as Config<I>>::WeightInfo::approve_bounty_with_curator())]
+		pub fn approve_bounty_with_curator(
+			origin: OriginFor<T>,
+			#[pallet::compact] bounty_id: BountyIndex,
+			curator: AccountIdLookupOf<T>,
+			#[pallet::compact] fee: BalanceOf<T, I>,
+		) -> DispatchResult {
+			let max_amount = T::SpendOrigin::ensure_origin(origin)?;
+			let curator = T::Lookup::lookup(curator)?;
+			Bounties::<T, I>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				// approve bounty
+				let bounty = maybe_bounty.as_mut().ok_or(Error::<T, I>::InvalidIndex)?;
+				ensure!(
+					bounty.value <= max_amount,
+					pallet_treasury::Error::<T, I>::InsufficientPermission
+				);
+				ensure!(bounty.status == BountyStatus::Proposed, Error::<T, I>::UnexpectedStatus);
+				ensure!(fee < bounty.value, Error::<T, I>::InvalidFee);
+
+				BountyApprovals::<T, I>::try_append(bounty_id)
+					.map_err(|()| Error::<T, I>::TooManyQueued)?;
+
+				bounty.status = BountyStatus::ApprovedWithCurator { curator: curator.clone() };
+				bounty.fee = fee;
+
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::<T, I>::BountyApproved { index: bounty_id });
+			Self::deposit_event(Event::<T, I>::CuratorProposed { bounty_id, curator });
```

Comment:
Do we need to emit both events? Is `BountyApproved` inferable from `CuratorProposed`?

Reply:
I think so, yes, because this function does logically what `approve_bounty` and `propose_curator` does, and they emit these events separately

Reply:
```
I’m not very familiar with the code, but having multiple statuses with similar meanings seems like a potential code smell.

How about modifying Approved with the field maybe_curator: Option<AccountId>? Probably we want to avoid any breaking change (which is totally understandable)? Though any indexer now needs to support the ApprovedWithCurator status as well, so I guess breaking change in this scenario is better.

I leave the judgement to you guys. Though it would be great to have some doc explaining expected lifecycle of a bounty (I couldn't find one).
```
Wouldn't that break existing which used `Approved` enum? I think it would add at least 1 byte for the option of `Approved` whether it is `None` or `Some`, with these changes we leave the rest of the enums unmodified and people would only need to add support for the new one

Reply:
> I think we need to emit both, because this function does logically what approve_bounty and propose_curator does, and they emit these events separately.

Only ask this since events are expensive and stored in state of each block. But of course if you have a good reason to emit both, go for it.

> Wouldn't maybe_curator: Option<AccountId> break existing which used Approved enum?

Is breaking change always bad? Especially since old clients may have unexpected side effects, i.e. just miss the approved (`ApprovedWithCurator`) status altogether if they depend on the existing `Approved` status. 

Also types are dynamically generated from substrate metadata and potentially for js clients this might not even be a breaking change.

P.S.: None of this is a huge deal tbh. I’m comfortable with either approach, as long as we’re mindful of the tradeoffs.